logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/SendKnockUseCase.kt

```diff
@@ -68,16 +71,29 @@ class SendKnockUseCase internal constructor(
                 content = MessageContent.Knock(hotKnock),
                 conversationId = conversationId,
                 date = DateTimeUtil.currentIsoDateTimeString(),
-                senderUserId = selfUser.id,
+                senderUserId = selfUserId,
                 senderClientId = currentClientId,
                 status = Message.Status.PENDING,
                 editStatus = Message.EditStatus.NotEdited,
                 isSelfMessage = true
             )
             persistMessage(message)
-        }.flatMap {
-            messageSender.sendPendingMessage(conversationId, generatedMessageUuid)
+                .flatMap {
+                    messageSender.sendMessage(message)
+                }
+                .onSuccess {
+                    messageRepository.updateMessageStatus(MessageEntity.Status.SENT, conversationId, generatedMessageUuid)
+                }
         }.onFailure {
```

Comment:
I see that this handler is similar to SendEditTextMessageUseCase, can we somehow extract it to separate function?

Reply:
Good idea, I moved the common code to the `MessageSendFailureHandler`.